### ISSUE TITLE
File upload to quickbooks doesn't work in production

### ISSUE THREAD

**Issue by rudresh4** [rudresh4/patchwork#141]
- File upload when loads are invoiced works in local environment but fails in production

```
I, [2017-12-01T11:19:26.902690 #23581]  INFO -- : BODY PART:
I, [2017-12-01T11:19:26.902718 #23581]  INFO -- : file_metadata_0: <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Attachable xmlns="http://schema.intuit.com/finance/v3" sparse="false">
<AttachableRef>
<EntityRef type="Invoice">11</EntityRef>
<IncludeOnSend>true</IncludeOnSend>
</AttachableRef>
<FileName>lumper20171201-23581-yblvhw.pdf</FileName>
<ContentType>application/pdf</ContentType>
</Attachable>
I, [2017-12-01T11:19:26.902748 #23581]  INFO -- : REQUEST HEADERS = {"Content-Type"=>"multipart/form-data", "Accept"=>"application/xml", "Accept-Encoding"=>"gzip, deflate"}
E, [2017-12-01T11:19:27.023209 #23581] ERROR -- : undefined method `bytesize' for #<Hash:0x0000562bf790dc78>
E, [2017-12-01T11:19:27.023299 #23581] ERROR -- : /opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http/generic_request.rb:183:in `send_request_with_body'
/opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http/generic_request.rb:121:in `exec'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/aws-sdk-core-3.9.0/lib/seahorse/client/net_http/patches.rb:28:in `block in new_transport_request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/aws-sdk-core-3.9.0/lib/seahorse/client/net_http/patches.rb:27:in `catch'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/aws-sdk-core-3.9.0/lib/seahorse/client/net_http/patches.rb:27:in `new_transport_request'
/opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http.rb:1416:in `request'
/opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http.rb:1409:in `block in request'
/opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http.rb:877:in `start'
/opt/rubies/ruby-2.4.2/lib/ruby/2.4.0/net/http.rb:1407:in `request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/adapter/net_http.rb:80:in `perform_request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/adapter/net_http.rb:38:in `block in call'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/adapter/net_http.rb:85:in `with_net_http_connection'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/adapter/net_http.rb:33:in `call'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/request/url_encoded.rb:15:in `call'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/rack_builder.rb:141:in `build_response'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/faraday-0.12.2/lib/faraday/connection.rb:386:in `run_request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/oauth2-1.4.0/lib/oauth2/client.rb:99:in `request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/gems/oauth2-1.4.0/lib/oauth2/access_token.rb:107:in `request'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/util/multipart.rb:83:in `multipart_post'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/util/multipart.rb:79:in `post_with_multipart'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/service/base_service.rb:295:in `oauth_post_with_multipart'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/service/base_service.rb:263:in `do_http'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/service/base_service.rb:234:in `do_http_file_upload'
/opt/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bundler/gems/quickbooks-ruby-e2d9715b4288/lib/quickbooks/service/upload.rb:13:in `upload'
```

**Comment by rudresh4** [comment]
- https://github.com/lostisland/faraday/issues/185
- https://github.com/lostisland/faraday/issues/291
- https://github.com/lostisland/faraday/issues/161

**Comment by rudresh4** [comment]
In development, the correct Faraday object is being sent

```
"#<Faraday::Env @method=:post @body=#<Faraday::CompositeReadIO:0x007fd8117bf7f0 @parts=[#<Parts::FilePart:0x007fd818be8028 @head="-------------RubyMultipartPost-08fb0e509bb64a5cf667311c9d5a3c3d\r\nContent-Disposition: form-data; name=\"file_content_0\"; 
```

But in production, only a hash is being sent in body:
```
#<Faraday::Env @method=:post @body={"file_content_0"=>#<UploadIO:0x000055a7ce3c0110 @content_type="application/pdf", @original_filename="52-
```

**Comment by rudresh4** [comment]
Fixed: 28a96aaede8de4cbf26f771c37d2e1e520e188e5
